stravalib/model.py

```diff
@@ -17,28 +22,105 @@
     META,
     SUMMARY,
     Attribute,
-    DateAttribute,
     EntityAttribute,
     EntityCollection,
     LocationAttribute,
     TimeIntervalAttribute,
     TimestampAttribute,
-    TimezoneAttribute,
 )
 from stravalib.exc import warn_method_deprecation
-from stravalib.field_conversions import enum_values, time_interval
+from stravalib.field_conversions import enum_values, time_interval, timezone
 from stravalib.strava_model import (
     ActivityStats,
     ActivityTotal,
+    ActivityType,
     Comment,
+    DetailedActivity,
     DetailedAthlete,
     DetailedClub,
     DetailedGear,
+    DetailedSegment,
+    DetailedSegmentEffort,
+    ExplorerSegment,
+    Lap,
+    LatLng,
     PhotosSummary,
+    PolylineMap,
     Primary,
+    Split,
+    SummaryPRSegmentEffort,
+    SummarySegmentEffort,
 )
 from stravalib.unithelper import UnitConverter
 
+LOGGER = logging.getLogger(__name__)
+
+
+def lazy_property(fn):
```

Comment:
This decorator checks if an entity has a bound client, whether the entity has an `id` field, etc. All these checks were repeated in the legacy model.